src/Relude/Extra/Foldable1.hs

```diff
@@ -0,0 +1,54 @@
+module Relude.Extra.Foldable1
+    ( Foldable1 (..)
+    , traverse1_
+    , for1_
+    ) where
+
+import Relude.Applicative (Applicative, pass)
+import Relude.Base (Ord)
+import Relude.Foldable.Reexport (Foldable (foldr))
+import Relude.Function (flip, id, (.))
+import Relude.Functor.Reexport (Compose (..), Identity (..))
+import Relude.Monoid (Semigroup, (<>))
+
+import Control.Applicative ((*>))
+import Data.List.NonEmpty (NonEmpty (..))
+import Data.Semigroup (First (..), Last (..), Max (..), Min (..))
+
+class Foldable t => Foldable1 t where
+    foldMap1   :: Semigroup m => (a -> m) -> t a -> m
+
+    fold1 :: Semigroup m => t m -> m
+    fold1 = foldMap1 id
+
+    toNonEmpty :: t a -> NonEmpty a
+    toNonEmpty = foldMap1 (:|[])
+
+    head1 :: t a -> a
+    head1 = getFirst . foldMap1 First
```

Comment:
It's better to use trick with `#.` operator from `Foldable` typeclass. See `minimum` function for example here:

* https://hackage.haskell.org/package/base-4.11.1.0/docs/src/Data.Foldable.html#minimum

This will improve efficiency of the code.